src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/ChainPal.cs

```diff
@@ -35,7 +38,319 @@ public static bool ReleaseSafeX509ChainHandle(IntPtr handle)
             TimeSpan timeout,
             bool disableAia)
         {
-            throw new NotImplementedException(nameof(BuildChain));
+            var chainPal = new AndroidCertPath();
+            try
+            {
+                chainPal.Initialize(cert, extraStore, customTrustStore, trustMode);
+                chainPal.Evaluate(verificationTime, applicationPolicy, certificatePolicy, revocationMode, revocationFlag);
+            }
+            catch
+            {
+                chainPal.Dispose();
+                throw;
+            }
+
+            return chainPal;
+        }
+
+        private sealed class AndroidCertPath : IChainPal
+        {
+            public X509ChainElement[]? ChainElements { get; private set; }
+            public X509ChainStatus[]? ChainStatus { get; private set; }
+
+            public SafeX509ChainHandle? SafeHandle => null;
+
+            private SafeX509ChainContextHandle? _chainContext;
+            private bool _isValid;
+
+            public void Dispose()
+            {
+                if (_chainContext != null)
+                {
+                    _chainContext.Dispose();
+                }
+            }
+
+            public bool? Verify(X509VerificationFlags flags, out Exception? exception)
+            {
+                Debug.Assert(_chainContext != null);
+                exception = null;
+
+                if (!_isValid)
+                {
+                    // There is no way to bypass time, trusted root, name, or policy constraint
+                    // validation on Android. It will not build any chain without these all
+                    // being valid.
+                    X509VerificationFlags[] unsupportedFlags = new X509VerificationFlags[]
```

Comment:
This should probably be a static readonly so it doesn't have to get recreated each time.  (`private static readonly X509VerificationFlags[] s_unsupportedVerificationFlags = ...`)